pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -230,11 +249,19 @@ func (state PipelineRunState) IsStopping(d *dag.Graph) bool {
 func (state PipelineRunState) GetNextTasks(candidateTasks sets.String) []*ResolvedPipelineRunTask {
 	tasks := []*ResolvedPipelineRunTask{}
 	for _, t := range state {
-		if _, ok := candidateTasks[t.PipelineTask.Name]; ok && t.TaskRun == nil {
+		if _, ok := candidateTasks[t.PipelineTask.Name]; ok && t.TaskRun == nil && t.Run == nil {
 			tasks = append(tasks, t)
 		}
-		if _, ok := candidateTasks[t.PipelineTask.Name]; ok && t.TaskRun != nil {
-			status := t.TaskRun.Status.GetCondition(apis.ConditionSucceeded)
+		if _, ok := candidateTasks[t.PipelineTask.Name]; ok {
```

Comment:
I find the following easier to follow that current code.

```go
if _, ok := candidateTasks[t.PipelineTask.Name]; ok {
	if t.TaskRun == nil && t.Run == nil {
		tasks = append(tasks, t)
	}
	// […]
}
```

or at least, extracting `_, ok := candidateTasks[t.PipelineTask.Name]` into its own variable and using it in the if or a switch/case :upside_down_face: